src/storage/state.rs

```diff
@@ -259,3 +287,37 @@ impl From<&HostHeartbeat> for Host {
         }
     }
 }
+
+// Custom Hash impl's due to nested HashMaps and HashSets
+impl Hash for WadmActorInstance {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.instance_id.hash(state);
+    }
+}
+
+/// NOTE(brooksmtownsend): This is compared only by the instance ID
+/// as it's a GUID and lookups don't require annotations
+impl PartialEq for WadmActorInstance {
+    fn eq(&self, other: &Self) -> bool {
+        self.instance_id == other.instance_id
+    }
+}
+impl Eq for WadmActorInstance {}
+
+impl Hash for Actor {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.id.hash(state);
+        self.name.hash(state);
+        self.capabilities.hash(state);
+        self.issuer.hash(state);
+        self.call_alias.hash(state);
+        self.reference.hash(state);
+
+        for (host_id, instances) in &self.instances {
```

Comment:
Same thing here with non deterministic hashing. However, does `Actor` need to be `Hash`? We aren't storing it in a HashSet anywhere, are we?

Reply:
Huh, looks like we aren't. Will remove